### PR TITLE
feat(cli): implement --resolver flag

### PR DESCRIPTION
**Pull request by P0lip**
Closes https://github.com/stoplightio/spectral/issues/717

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/cli/services/linter/utils/getResolver.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { isAbsolute, join } from '@stoplight/path';
+import { Optional } from '@stoplight/types';
+import { httpAndFileResolver } from '../../../../resolvers/http-and-file';
+
+export const getResolver = (resolver: Optional<string>) => {
+  if (resolver !== void 0) {
+    try {
+      return require(isAbsolute(resolver) ? resolver : join(process.cwd(), resolver));
```
[the remark this answers is not in the document]
Comment (P0lip) [reply src/cli/services/linter/utils/getResolver.ts]: This code is not supposed to be executed by JS API ever, so Studio should not be affected.
It's used only by the CLI part of Spectral.

File: src/cli/services/linter/utils/getResolver.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { isAbsolute, join } from '@stoplight/path';
+import { Optional } from '@stoplight/types';
+import { httpAndFileResolver } from '../../../../resolvers/http-and-file';
+
+export const getResolver = (resolver: Optional<string>) => {
+  if (resolver !== void 0) {
```
[the remark this answers is not in the document]
Comment (P0lip) [reply src/cli/services/linter/utils/getResolver.ts]: What if string is empty?

[the remark this answers is not in the document]
Comment (marbemac) [reply src/cli/services/linter/utils/getResolver.ts]: If string is empty we probably don't want to proceed right? Aka ignore it.

[the remark this answers is not in the document]
Comment (P0lip) [reply src/cli/services/linter/utils/getResolver.ts]: Well, right. I'll make it `if (resolver)`
```suggestion
  if (resolver) {
```

**Pull request merged**